### PR TITLE
Add tasks and metadata properties to python ControlMessage

### LINKED ISSUE #1131: [FEA]: Add `tasks` and `metadata` readonly properties to `ControlMessage`
### Is this a new feature, an improvement, or a change to existing functionality?

Improvement

### How would you describe the priority of this feature request

Low (would be nice)

### Please provide a clear description of problem this feature solves

It can be difficult to debug `ControlMessages` because you cannot see or enumerate the tasks on the message object. Being able to iterate over the tasks or view the task dictionary would be helpful.

### Describe your ideal solution

Add new, readonly properties `tasks` and `metadata` to the `ControlMessage` object with bindings to allow them to be used from python.

These properties should make copies of any object to prevent direct manipulation of the underlying data.

### Describe any alternatives you have considered

_No response_

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I have searched the [open feature requests](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3A%22feature+request%22%2Cimprovement%2Cenhancement) and have found no duplicates for this feature request

### PR DESCRIPTION
**Pull request by cwharris**
Adding `tasks` and `metadata` properties to python `ControlMessage`, allowing for easier debugging of control messages.

Closes #1131

### CODE REVIEW

File: morpheus/_lib/include/morpheus/messages/control.hpp [hunk 1]
```diff
@@ -68,6 +68,16 @@ class ControlMessage
      */
     void add_task(const std::string& task_type, const nlohmann::json& task);
 
+    /**
+     * @brief Get the tasks for the control message.
+     */
+    const nlohmann::json& tasks();
```
Comment (mdemoret-nv) [inline morpheus/_lib/include/morpheus/messages/control.hpp:74]:
```suggestion
    const nlohmann::json& tasks() const;
```
Should be `const`, right? Dont want to allow editing this directly outside of the `add`/`remove` functions

File: morpheus/_lib/include/morpheus/messages/control.hpp [hunk 2]
```diff
@@ -68,6 +68,16 @@ class ControlMessage
...
+
+    /**
+     * @brief Get the metadata for the control message.
+     */
+    const nlohmann::json& metadata();
```
Comment (mdemoret-nv) [inline morpheus/_lib/include/morpheus/messages/control.hpp:79]:
```suggestion
    const nlohmann::json& metadata() const;
```
Should be `const`, right? Dont want to allow editing this directly outside of the `add`/`remove` functions

**Review by mdemoret-nv — APPROVED** [review]
Just 2 changes

File: morpheus/_lib/include/morpheus/messages/control.hpp [hunk 3]
```diff
@@ -180,7 +180,7 @@ struct ControlMessageProxy
      * @param value
      */
     static void set_metadata(ControlMessage& self, const std::string& key, pybind11::object& value);
-    static pybind11::object get_metadata(ControlMessage& self, const std::string& key);
+    static pybind11::object get_metadata(ControlMessage& self, std::optional<std::string> const& key);
```
Comment (mdemoret-nv) [inline morpheus/_lib/include/morpheus/messages/control.hpp:183]: This should return a `pybind11::dict`

**Review by mdemoret-nv — APPROVED** [review]
Just one suggestion to help with typing

File: morpheus/_lib/include/morpheus/messages/control.hpp (hunk 3 above)
Comment (cwharris) [reply morpheus/_lib/include/morpheus/messages/control.hpp:183]: unfortunately this function can return either a `dict` or the value pointed to by the metadata key, so I'm not sure we can change this.

**Review by mdemoret-nv — APPROVED** [review]

**Pull request merged**